src/ray/gcs/gcs_server/gcs_placement_group_scheduler.cc

```diff
@@ -205,5 +224,24 @@ GcsPlacementGroupScheduler::GetOrConnectLeaseClient(const rpc::Address &raylet_a
   return iter->second;
 }
 
+std::shared_ptr<ScheduleContext> GcsPlacementGroupScheduler::GetScheduleContext(
```

Comment:
I think `unique_ptr` is fine? Probably not a big deal because this code path is not performance sensitive.

Reply:
fixed